src/controllers/deploy.go

```diff
@@ -69,7 +69,11 @@ func internaldeploy(a *history.ActionInstance) ([]byte, error) {
 		state.Access = a.Access
 		state.Server = a.Server
 		state.Status = "deploying"
-		history.SetState(a.RepoURL, state)
+		if err1 := history.SetState(a.RepoURL, state); err1 != nil {
+			log.Infof("subdomain(%s) in use", a.RepoURL)
```

Comment:
You should not log an error message, which has more context than the present caller.
So it should be
`log.Infof('cannot set state to deploying - %v', err1)
This way we do not need to change error messages when we add some new functionality to history.SetSate()